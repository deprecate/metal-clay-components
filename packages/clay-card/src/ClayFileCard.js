import './ClayCard';
import Component from 'metal-component';
import defineWebComponent from 'metal-web-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './ClayFileCard.soy.js';

/**
 * Metal ClayFileCard component.
 */
class ClayFileCard extends Component {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayFileCard.STATE = {
	/**
	 * Actions in the dropdown on card.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?array|undefined}
	 * @default undefined
	 */
	actionItems: Config.arrayOf(
		Config.shapeOf({
			label: Config.string().required(),
			separator: Config.bool().value(false),
			url: Config.string(),
		})
	),

	/**
	 * Flag to indicate if the card is disabled or not.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?bool|undefined}
	 * @default false
	 */
	disabled: Config.bool().value(false),

	/**
	 * CSS classes to be applied to the element.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?string|undefined}
	 * @default undefined
	 */
	elementClasses: Config.string(),

	/**
	 * Type of the file represented in the card.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?string|undefined}
	 * @default undefined
	 */
	fileType: Config.string(),

	/**
	 * Style of type of the file represented in the card.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?string}
	 * @default primary
	 */
	fileTypeStyle: Config.oneOf([
		'danger',
		'dark',
		'info',
		'light',
		'primary',
		'secondary',
		'success',
		'warning',
	]).value('primary'),

	/**
	 * Render ClayIcon in the background element.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?string|undefined}
	 * @default documents-and-media
	 */
	icon: Config.string().value('documents-and-media'),

	/**
	 * Id to be applied to the element.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?string|undefined}
	 * @default undefined
	 */
	id: Config.string(),

	/**
	 * Name to be applied to the input element.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?string|undefined}
	 * @default undefined
	 */
	inputName: Config.string(),

	/**
	 * Value to be applied to the input element.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?string|undefined}
	 * @default undefined
	 */
	inputValue: Config.string(),

	/**
	 * Labels of the card.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?array|undefined}
	 * @default undefined
	 */
	labels: Config.arrayOf(
		Config.shapeOf({
			label: Config.string(),
			style: Config.oneOf([
				'danger',
				'info',
				'secondary',
				'success',
				'warning',
			]).value('secondary'),
		})
	),

	/**
	 * Flag to indicate if the card is selectable or not.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?bool|undefined}
	 * @default false
	 */
	selectable: Config.bool().value(false),

	/**
	 * Flag to indicate if the card is selected or not.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?bool|undefined}
	 * @default false
	 */
	selected: Config.bool().value(false),

	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {!string}
	 * @default undefined
	 */
	spritemap: Config.string().required(),

	/**
	 * Subtitle of the card.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?string|undefined}
	 * @default undefined
	 */
	subtitle: Config.string(),

	/**
	 * Title of the card.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?string|undefined}
	 * @default undefined
	 */
	title: Config.string(),

	/**
	 * Transform title in the url.
	 * @instance
	 * @memberof ClayFileCard
	 * @type {?string}
	 * @default undefined
	 */
	url: Config.string(),
};

defineWebComponent('clay-file-card', ClayFileCard);

Soy.register(ClayFileCard, templates);

export {ClayFileCard};
export default ClayFileCard;
