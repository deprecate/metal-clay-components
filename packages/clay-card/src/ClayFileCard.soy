{namespace ClayFileCard}

{template .render}
    {@param spritemap: string}
    {@param? actionItems: list<[
        label: string,
        separator: bool,
        url: string
    ]>}
    {@param? disabled: bool}
    {@param? elementClasses: string}
    {@param? fileType: string}
	{@param? fileTypeStyle: string}
    {@param? icon: string}
    {@param? id: string}
    {@param? inputName: string}
    {@param? inputValue: string}
	{@param? labels: list<[
		label: string,
		style: string
	]>}
    {@param? selectable: bool}
    {@param? selected: bool}
    {@param? subtitle: string}
    {@param? title: string}
    {@param? url: string}

    {call ClayCard.render}
        {param actionItems: $actionItems /}
        {param disabled: $disabled /}
        {param elementClasses: $elementClasses /}
        {param fileType: $fileType /}
        {param fileTypeStyle: $fileTypeStyle /}
        {param icon: $icon ?: 'documents-and-media' /}
        {param id: $id /}
        {param inputName: $inputName /}
        {param inputValue: $inputValue /}
        {param labels: $labels /}
        {param ref: 'cardFile' /}
        {param selectable: $selectable /}
        {param selected: $selected /}
        {param spritemap: $spritemap /}
        {param subtitle: $subtitle /}
        {param title: $title /}
        {param url: $url /}
        {param variant: 'file' /}
    {/call}
{/template}

/**
 * Render the visual area of the card.
 */
{deltemplate ClayCard.VisualArea variant="'file'"}
	{@param? icon: string}
	{@param? imageAlt: string}
	{@param? imageSrc: string}
	{@param? spritemap: string}
	{@param? fileType: string}
	{@param? fileTypeStyle: string}
	{@param? variant: string}

	<div class="aspect-ratio card-item-first">
		{if $icon and $spritemap}
			<div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
				{call ClayIcon.render}
					{param spritemap: $spritemap /}
					{param symbol: $icon /}
				{/call}
			</div>
		{/if}

		{if $fileType}
			{call ClaySticker.render}
				{param label: $fileType /}
				{param position: 'bottom-left' /}
				{param style: $fileTypeStyle ?: 'primary' /}
			{/call}
		{/if}
	</div>
{/deltemplate}