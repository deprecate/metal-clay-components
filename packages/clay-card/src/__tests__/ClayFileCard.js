import ClayFileCard from '../ClayFileCard';

let component;
let spritemap = 'icons.svg';
let actionItems = [
	{
		label: 'Edit',
		url: '#1',
	},
	{
		label: 'Save',
		url: '#1',
	},
];

describe('ClayFileCard', function() {
	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render the default markup', () => {
		component = new ClayFileCard({
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with classes', () => {
		component = new ClayFileCard({
			spritemap: spritemap,
			elementClasses: 'my-custom-class',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with id', () => {
		component = new ClayFileCard({
			spritemap: spritemap,
			id: 'myId',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with actionItems', () => {
		component = new ClayFileCard({
			actionItems: actionItems,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a selected ClayFileCard', () => {
		component = new ClayFileCard({
			selectable: true,
			selected: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a disabled ClayFileCard', () => {
		component = new ClayFileCard({
			disabled: true,
			selectable: true,
			selected: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with input `name`', () => {
		component = new ClayFileCard({
			inputName: 'checkbox01',
			selectable: true,
			selected: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with input `value`', () => {
		component = new ClayFileCard({
			inputValue: 'checkbox',
			selectable: true,
			selected: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with icon', () => {
		component = new ClayFileCard({
			icon: 'documents',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with label', () => {
		component = new ClayFileCard({
			labels: [
				{
					label: 'Approved',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with labelStyle `danger color`', () => {
		component = new ClayFileCard({
			labels: [
				{
					label: 'Approved',
					style: 'danger',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with labelStyle `info color`', () => {
		component = new ClayFileCard({
			labels: [
				{
					label: 'Approved',
					style: 'info',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with labelStyle `secondary color`', () => {
		component = new ClayFileCard({
			labels: [
				{
					label: 'Approved',
					style: 'secondary',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with labelStyle `success color`', () => {
		component = new ClayFileCard({
			labels: [
				{
					label: 'Approved',
					style: 'success',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with labelStyle `warning color`', () => {
		component = new ClayFileCard({
			labels: [
				{
					label: 'Approved',
					style: 'warning',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with subtitle', () => {
		component = new ClayFileCard({
			subtitle: 'Author Action',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with title', () => {
		component = new ClayFileCard({
			title: 'empty-background.png',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with file type', () => {
		component = new ClayFileCard({
			fileType: 'DOC',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with file type style `danger color`', () => {
		component = new ClayFileCard({
			fileType: 'DOC',
			fileTypeStyle: 'danger',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with file type style `dark color`', () => {
		component = new ClayFileCard({
			fileType: 'DOC',
			fileTypeStyle: 'dark',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with file type style `info color`', () => {
		component = new ClayFileCard({
			fileType: 'DOC',
			fileTypeStyle: 'info',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with file type style `light color`', () => {
		component = new ClayFileCard({
			fileType: 'DOC',
			fileTypeStyle: 'light',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with file type style `primary color`', () => {
		component = new ClayFileCard({
			fileType: 'DOC',
			fileTypeStyle: 'primary',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with file type style `secondary color`', () => {
		component = new ClayFileCard({
			fileType: 'DOC',
			fileTypeStyle: 'secondary',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with file type style `success color`', () => {
		component = new ClayFileCard({
			fileType: 'DOC',
			fileTypeStyle: 'success',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with file type style `warning color`', () => {
		component = new ClayFileCard({
			fileType: 'DOC',
			fileTypeStyle: 'warning',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayFileCard with url', () => {
		component = new ClayFileCard({
			url: '#1',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});
});
