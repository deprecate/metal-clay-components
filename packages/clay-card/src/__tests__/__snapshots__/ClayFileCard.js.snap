// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayFileCard should render a ClayFileCard with actionItems 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
      <div class="flex-col">
        <div class="dropdown dropdown-action">
          <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v">
              <use xlink:href="icons.svg#ellipsis-v"></use>
            </svg>
          </button>
          <div class="dropdown-menu">
            <ul class="list-unstyled">
              <li>
                <a class=" dropdown-item" href="#1">Edit</a>
              </li>
              <li>
                <a class=" dropdown-item" href="#1">Save</a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with classes 1`] = `
<div class="card-type-asset card my-custom-class">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with file type 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
    <span class="sticker sticker-primary sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with file type style \`danger color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
    <span class="sticker sticker-danger sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with file type style \`dark color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
    <span class="sticker sticker-dark sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with file type style \`info color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
    <span class="sticker sticker-info sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with file type style \`light color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
    <span class="sticker sticker-light sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with file type style \`primary color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
    <span class="sticker sticker-primary sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with file type style \`secondary color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
    <span class="sticker sticker-secondary sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with file type style \`success color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
    <span class="sticker sticker-success sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with file type style \`warning color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
    <span class="sticker sticker-warning sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with icon 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents">
        <use xlink:href="icons.svg#documents"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with id 1`] = `
<div class="card-type-asset card" id="myId">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with input \`name\` 1`] = `
<div class="card-type-asset form-check form-check-card form-check-top-left">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" name="checkbox01" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <use xlink:href="icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with input \`value\` 1`] = `
<div class="card-type-asset form-check form-check-card form-check-top-left">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" value="checkbox" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <use xlink:href="icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with label 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-secondary">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with labelStyle \`danger color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-danger">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with labelStyle \`info color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-info">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with labelStyle \`secondary color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-secondary">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with labelStyle \`success color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-success">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with labelStyle \`warning color\` 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-warning">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with subtitle 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-subtitle text-truncate" title="Author Action">Author Action</div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with title 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-title text-truncate" title="empty-background.png">empty-background.png</div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a ClayFileCard with url 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <a class=" card-title text-truncate" href="#1"></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayFileCard should render a disabled ClayFileCard 1`] = `
<div class="card-type-asset form-check form-check-card form-check-top-left">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" disabled="disabled" class="custom-control-input" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <use xlink:href="icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayFileCard should render a selected ClayFileCard 1`] = `
<div class="card-type-asset form-check form-check-card form-check-top-left">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
              <use xlink:href="icons.svg#documents-and-media"></use>
            </svg>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayFileCard should render the default markup 1`] = `
<div class="card-type-asset card">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents-and-media">
        <use xlink:href="icons.svg#documents-and-media"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;
